back/app/Http/Controllers/MessageController.php

```diff
@@ -0,0 +1,65 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use Illuminate\Http\Request;
+use App\Http\Requests\MessageRequest;
+use App\Http\Requests\UpdateMessageActivationRequest;
+use App\Models\Message;
+use App\Http\Resources\MessageResource;
+use Illuminate\Http\JsonResponse;
+
+class MessageController extends Controller
+{
+    public function index()
+    {
+        $this->authorize('index', Message::class);
+        return response()->json(MessageResource::collection(Message::all()));
+    }
+
+    public function store(MessageRequest $request)
+    {
+        $this->authorize('store', Message::class);
+        $message = Message::create(
+            $request->only(['title', 'content', 'link'])
+        );
+        $message->is_active = 0;
+        $message->save();
+
+        return response()->json(MessageResource::make($message), JsonResponse::HTTP_CREATED);
+    }
+
+    public function update(MessageRequest $request,$message_id)
+    {
+        $message = Message::findOrFail($message_id);
+        $this->authorize('update', $message);
+        $message->update($request->only(['title', 'content', 'link']));
+
+        return response()->json(MessageResource::make($message), JsonResponse::HTTP_CREATED);
+    }
+
+    public function updateActivated(UpdateMessageActivationRequest $request, $message_id)
+    {
+        $all_messages = Message::all();
+        $all_messages->each(function($all_message)
+        {
+            $all_message->is_active = 0;
+            $all_message->save();
+        });
+        $message = Message::findOrFail($message_id);
+        $this->authorize('updateActivated', $message);
+        $message->update(['is_active' => $request['is_active']]);
+
+        return response()->json(MessageResource::make($message), JsonResponse::HTTP_CREATED);
```

Comment:
L'idée est bonne ici, c'est presque ce qu'il faut faire. Une chose importante c'est le moment ou tu appelles la méthode `$this->authorize(...)` comme je t'avais dis. Il faut que tu appelles cette fonction avant TOUTE opération dans la bdd, sinon l'utilisateur va modifier des entrées en bdd AVANT que l'authorisation ait lieu, et c'est pas ce qu'on veut. On veut qu'il ne modifie des choses que s'il a le droit de le faire, et donc c'est pour ça que l'appelle à `$this->authorize(...)` doit se faire en premier lieu.

Pour ce qui est du code, ici tu désactives tous les messages dans la bdd peu importe s'ils sont actifs ou pas. Le mieux serait de désactiver uniquement ceux qui sont actifs : 
```suggestion
        $message = Message::findOrFail($message_id);
        $this->authorize('updateActivated', $message);
        
        Message::active()->update(['is_active', false]);
        $message->update(['is_active' => $request['is_active']]);

        return response()->json(MessageResource::make($message), JsonResponse::HTTP_CREATED);
```